File: Easy/Pascal Triangle/pascal-triangle.py
#User function Template for python3

#User function Template for python3
class Solution:

    def nthRowOfPascalTriangle(self,n):
	    ans=1
        mod=10**9+7
        l2=[]
        l1=[1]
        if(n==1):
            return l1
        elif(n==2):
            return l1+l1
        for i in range(1,n):
            ans=ans*(n-i)
            ans=ans//i
            l2.append(ans%mod)
        return l1+l2


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 

	tc=int(input())
	while tc > 0:
	    n=int(input())
	    ob = Solution()
	    ans=ob.nthRowOfPascalTriangle(n)
	    for x in ans:
	        print(x, end=" ")
	    print()
	    tc=tc-1
# } Driver Code Ends

File: Difficulty: Medium/Pascal Triangle/pascal-triangle.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends



class Solution {
  public:
    vector<int> nthRowOfPascalTriangle(int n) {
        // code here
        n=n-1;
        vector<int> row;
        long long val = 1;
        for (int k = 0; k <= n; ++k) {
            row.push_back(val);
            val = val * (n - k) / (k + 1);
        }
        return row;
    }
};


//{ Driver Code Starts.

void printAns(vector<int> &ans) {
    for (auto &x : ans) {
        cout << x << " ";
    }
    cout << "\n";
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution ob;
        auto ans = ob.nthRowOfPascalTriangle(n);
        printAns(ans);

        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends